spell-check.R

```diff
@@ -31,11 +31,27 @@ if (file.exists(dict_file)) {
   dictionary <- ""
 }
 
-# Run spell check
-spelling_errors <- spelling::spell_check_files(files, ignore = dictionary) |>
-  data.frame() |>
-  tidyr::unnest(cols = found) |>
-  tidyr::separate(found, into = c("file", "lines"), sep = ":")
+
+# Separate files into path groups
+all_paths <- files |>
+  dirname() |>
+  unique()
```

Comment:
Doing a second search for files later seems like a potential problem, so why not preserve these as groups:
```suggestion
file_list <- split(files, dirname(files))
```

Then you can use purrr::imap():
```
file_list |> purrr::imap(\(files, path){
  spelling::spell_check_files(files, ignore = dictionary) |>
 ...
```